### PR TITLE
feature/reward-delivery-confirmation-email

### PR DESCRIPTION
**Pull request by rniedziolka**
Hi,
the next feature is ready:
https://www.loom.com/share/34894173bbb94a219a1eb71ddf0b8217
Waiting for the feedback!

### CODE REVIEW

File: config/environments/development.rb [hunk 1]
```diff
@@ -44,7 +44,19 @@
   config.active_storage.service = :local
 
   # Don't care if the mailer can't send.
-  config.action_mailer.raise_delivery_errors = false
+  config.action_mailer.raise_delivery_errors = true
+  config.action_mailer.perform_deliveries = true
+  config.action_mailer.delivery_method = :smtp
```
Comment (ciejj) [inline config/environments/development.rb]: You should rather use letter_opener for the dev env
```suggestion
config.action_mailer.delivery_method = :letter_opener
```

File: spec/mailers/order_delivery_mailer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe OrderDeliveryMailer, type: :mailer do
+  describe 'Order delivery confirmation' do
+    let(:order) { create(:order) }
+    let(:mail) { described_class.with(order: order).delivery_email.deliver_now }
+
+    it 'checking the headers' do
```
Comment (ciejj) [inline spec/mailers/order_delivery_mailer_spec.rb]: suggestion for naming specs
```suggestion
    it 'has proper headers' do
```

File: spec/mailers/order_delivery_mailer_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      expect(mail.subject).to have_content 'Your order has been delivered'
+      expect(mail.to).to have_content order.employee.email
+      expect(mail.from).to have_content 'contact@graphn.eu'
+    end
+
+    it 'checking the body' do
```
Comment (ciejj) [inline spec/mailers/order_delivery_mailer_spec.rb]:
```suggestion
    it 'has proper body' do
```

File: app/mailers/order_delivery_mailer.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class OrderDeliveryMailer < ApplicationMailer
+  def delivery_email
```
Comment (ciejj) [inline app/mailers/order_delivery_mailer.rb]: I find this name a bit misleading, what would you say about `delivery_confirmation_email` instead?

File: spec/mailers/previews/order_delivery_preview.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class OrderDeliveryPreview < ActionMailer::Preview
+  def delivery_email
+    OrderDeliveryMailer.with(order: order).delivery_email
```
Comment (ciejj) [inline spec/mailers/previews/order_delivery_preview.rb]: is it really working? how are you passing the `order`?

**Review by ciejj — COMMENTED** [review]
Well done, we are almost there, but I think you can slightly improve it :)

File: spec/mailers/previews/order_delivery_preview.rb (hunk 5 above)
Comment (rniedziolka) [reply spec/mailers/previews/order_delivery_preview.rb]: yes, this is model defined in factories
this returns order reward title in e-mail to employee

File: app/mailers/order_delivery_mailer.rb (hunk 4 above)
Comment (rniedziolka) [reply app/mailers/order_delivery_mailer.rb]: ok, fixed

File: spec/mailers/order_delivery_mailer_spec.rb (hunk 3 above)
Comment (rniedziolka) [reply spec/mailers/order_delivery_mailer_spec.rb]: my mistake, fixed

File: spec/mailers/order_delivery_mailer_spec.rb (hunk 2 above)
Comment (rniedziolka) [reply spec/mailers/order_delivery_mailer_spec.rb]: fixed

File: config/environments/development.rb (hunk 1 above)
Comment (rniedziolka) [reply config/environments/development.rb]: I put this in dev env because I can send e-mail confirmation without sending assets to heroku.
Fixed now.

**Comment by rniedziolka** [comment]
Hi,
@ciejj - fixes: https://github.com/rniedziolka/employee-recognition-platform-1637255604/pull/26/commits/19b66113e276687351688d80d4565c740e93c40c 
Thank you very much!

File: spec/mailers/previews/order_delivery_preview.rb (hunk 5 above)
Comment (ciejj) [reply spec/mailers/previews/order_delivery_preview.rb]: could you please attach a screenshot of that preview?

Comment (rniedziolka) [reply spec/mailers/previews/order_delivery_preview.rb]: Screenshot 2022-04-13 at 11 52 53

Comment (ciejj) [reply spec/mailers/previews/order_delivery_preview.rb]: where does the "Cauliflower Penne" comes from?
[edit] what I mean, for me it looks like the value from the seeds, not factory - so it is not a preview, but email, that has been sent upon the delivery

Comment (rniedziolka) [reply spec/mailers/previews/order_delivery_preview.rb]: Yes, you are right, fixed: https://github.com/rniedziolka/employee-recognition-platform-1637255604/pull/26/commits/6fb5ecf36a2a620c1b7dfc9a49a73eb67cfd7db1
Screenshot 2022-04-13 at 12 29 34

**Review by ciejj — APPROVED** [review]
👍

**Pull request merged**